Tasks/SonarQubePostTest/PRCA/ReportProcessor.ps1

```diff
@@ -2,6 +2,80 @@ $ProjectGuidAndFilePathMap = @{}
 $ComponentKeyAndPathMap = @{}
 $ComponentKeyAndRelativePathCache = @{}
 
+
+#region Public
+
+#
+# Fetches the new issues from the json report and also adds a "relativePath" for easier consumption
+#
+# Remark: the SonarQube report uses the component guid to refer to paths
+#
+function FetchAnnotatedNewIssues
+{    
+    $sonarReportFilePath = GetSonarReportFilePath
+    if (![System.IO.File]::Exists($sonarReportFilePath))
+    {
+        throw "Could not find the SonarQube issue report at $sonarReportFilePath. Unable to post issues to the PR."
```

Comment:
I'm now throwing exceptions for all missing files / empty env variables etc. Fail fast. Note that users have a mechanism of disabling SQ analysis on PR builds, should anything go wrong.